service/src/main/java/bio/terra/workspace/db/model/DbUpdater.java

```diff
@@ -0,0 +1,204 @@
+      if (dbCloningInstructions == null) {
+        return null;
+      }
+      return CloningInstructions.fromSql(dbCloningInstructions);
+    }
+
+    @JsonIgnore
+    public UpdateData setCloningInstructions(@Nullable CloningInstructions cloningInstructions) {
+      if (cloningInstructions == null) {
+        this.dbCloningInstructions = null;
+      } else {
+        this.dbCloningInstructions = cloningInstructions.toSql();
+      }
+      return this;
+    }
+
+    // Get/set for serdes
+    public String getDbCloningInstructions() {
+      return dbCloningInstructions;
+    }
+
+    public UpdateData setDbCloningInstructions(String dbCloningInstructions) {
+      this.dbCloningInstructions = dbCloningInstructions;
+      return this;
+    }
+
+    public @Nullable String getJsonAttributes() {
+      return jsonAttributes;
+    }
+
+    public UpdateData setJsonAttributes(@Nullable String jsonAttributes) {
+      this.jsonAttributes = jsonAttributes;
+      return this;
+    }
+  }
+
+  // We maintain the original data and any updated data
+  private final UpdateData originalData;
+  private final UpdateData updatedData;
+
+  /**
+   * This constructor and the two getters are only intended for use by Jackson JSON serdes.
+   *
+   * @param originalData original data
+   * @param updatedData any updated to original data
+   */
+  @JsonCreator
+  public DbUpdater(
+      @JsonProperty("originalData") UpdateData originalData,
+      @JsonProperty("updatedData") UpdateData updatedData) {
+    this.originalData = originalData;
+    this.updatedData = updatedData;
+  }
+
+  public UpdateData getOriginalData() {
+    return originalData;
+  }
+
+  public UpdateData getUpdatedData() {
+    return updatedData;
+  }
+
+  /**
+   * Used by ResourceDao to construct a DbUpdater
+   *
+   * @param name resource name
+   * @param description resource description
+   * @param cloningInstructions cloning instructions
+   * @param jsonAttributes JSON string form of the resource attributes
+   */
+  public DbUpdater(
+      String name,
+      @Nullable String description,
+      @Nullable CloningInstructions cloningInstructions,
+      @Nullable String jsonAttributes) {
+    this.originalData =
+        new UpdateData()
+            .setName(name)
+            .setDescription(description)
+            .setCloningInstructions(cloningInstructions)
+            .setJsonAttributes(jsonAttributes);
+    this.updatedData =
+        new UpdateData()
+            .setName(null)
+            .setDescription(null)
+            .setCloningInstructions(null)
+            .setJsonAttributes(null);
+  }
+
+  /**
+   * Populate the updated data using the CommonUpdateParameters gather from the API input of an
+   * update endpoint.
+   *
+   * @param commonUpdateParameters class containing the updated parameters
+   */
+  public void updateFromCommonParameters(CommonUpdateParameters commonUpdateParameters) {
+    if (commonUpdateParameters.getName() != null) {
```

Comment:
what if `commonUpdateParameters` is null?

Reply:
Then there will be a NPE. The interface contract is not nullable.
What would I do if it were null? Throw some exception other than NPE?